code/modules/mob/living/basic/minebots/minebot.dm

```diff
@@ -0,0 +1,191 @@
+	for(var/obj/item/stack/ore/ore_stack in contents)
+		ore_count += ore_stack.amount
+
+	. += "<span class='notice'>It is currently storing [ore_count] ore.</span>"
+
+	if(stored_gun && stored_gun.max_mod_capacity)
+		. += "<span class='notice'><b>[stored_gun.get_remaining_mod_capacity()]%</b> mod capacity remaining.</span>"
+		for(var/A in stored_gun.get_modkits())
+			var/obj/item/borg/upgrade/modkit/M = A
+			. += "<span class='notice'>There is \a [M] installed, using <b>[M.cost]%</b> capacity.</span>"
+
+/mob/living/basic/mining_drone/item_interaction(mob/living/user, obj/item/I, list/modifiers)
+	if(istype(I, /obj/item/mining_scanner) || istype(I, /obj/item/t_scanner/adv_mining_scanner))
+		to_chat(user, "<span class='notice'>You instruct [src] to drop any collected ore.</span>")
+		DropOre()
+		return ITEM_INTERACT_COMPLETE
+	if(istype(I, /obj/item/borg/upgrade/modkit))
+		I.melee_attack_chain(user, stored_gun, list2params(modifiers))
+		return ITEM_INTERACT_COMPLETE
+
+	return ..()
+
+/mob/living/basic/mining_drone/crowbar_act(mob/user, obj/item/I)
+	if(user.a_intent != INTENT_HELP)
+		return
+	. = TRUE
+	if(!I.tool_start_check(src, user, 0))
+		return
+	I.melee_attack_chain(user, stored_gun)
+
+/mob/living/basic/mining_drone/welder_act(mob/user, obj/item/I)
+	if(user.a_intent != INTENT_HELP)
+		return
+	. = TRUE
+	if(health == maxHealth)
+		to_chat(user, "<span class='notice'>[src] doesn't need repairing!</span>")
+		return
+	if(!I.tool_use_check(user, 1))
+		return
+	WELDER_ATTEMPT_REPAIR_MESSAGE
+	if(I.use_tool(src, user, 15, 1, volume = I.tool_volume) && health != maxHealth)
+		adjustBruteLoss(-20)
+		WELDER_REPAIR_SUCCESS_MESSAGE
+	return
+
+/mob/living/basic/mining_drone/death()
+	DropOre(0)
+	if(stored_gun)
+		for(var/obj/item/borg/upgrade/modkit/M in stored_gun.modkits)
+			M.uninstall(stored_gun)
+	robogibs(loc)
+	deathmessage = "blows apart!"
+	. = ..()
+
+/mob/living/basic/mining_drone/CanPass(atom/movable/O)
+	if(istype(O, /obj/item/projectile/kinetic))
+		var/obj/item/projectile/kinetic/K = O
+		if(K.kinetic_gun)
+			for(var/A in K.kinetic_gun.get_modkits())
+				var/obj/item/borg/upgrade/modkit/M = A
+				if(istype(M, /obj/item/borg/upgrade/modkit/minebot_passthrough))
+					return TRUE
+	if(istype(O, /obj/item/projectile/destabilizer))
+		return TRUE
+	return ..()
+
+/mob/living/basic/mining_drone/RangedAttack(atom/target, list/modifiers)
+	if(a_intent != INTENT_HARM)
+		return
+	stored_gun.afterattack__legacy__attackchain(target, src)
+
+/mob/living/basic/mining_drone/UnarmedAttack(atom/attack_target, proximity_flag, list/modifiers)
+	. = ..()
+
+	if(!. || !proximity_flag || (a_intent == INTENT_HARM))
+		return
+
+	if(istype(attack_target, /obj/item/stack/ore))
+		var/obj/item/target_ore = attack_target
+		target_ore.forceMove(src)
+
+/mob/living/basic/mining_drone/proc/CollectOre()
+	for(var/obj/item/stack/ore/O in range(1, src))
+		O.forceMove(src)
+
+/mob/living/basic/mining_drone/proc/DropOre(message = 1)
```

Comment:
Can you change the capitalization of these procs, and also make the default arguments use TRUE/FALSE instead of 1/0?

Reply:
Done.